acasclient/lsthing.py

```diff
@@ -2039,11 +2039,15 @@ def add_link(self, verb=None, linked_thing=None, recorded_by=None, metadata=None
         :type recorded_by: str, optional
         :param metadata: Dictionary of metadata to associate with the link itself, defaults to {}
         :type metadata: dict, optional
-        :param results: Dictioanry of results to associate with the link itself, defaults to {}
+        :param results: Dictionary of results to associate with the link itself, defaults to {}
         :type results: dict, optional
+        :param subject_type: The type of the subject of the link, defaults to the ls_type of itself
+        :type subject_type: str, optional
         """
-        self.links.append(SimpleLink(verb=verb, object=linked_thing, subject_type=self.ls_type,
-            recorded_by=recorded_by, metadata=metadata or {}, results=results or {}))
+        if not subject_type:
+            subject_type = self.ls_type
```

Comment:
subject_type continues to default to the current objects ls_type if not specified as a param